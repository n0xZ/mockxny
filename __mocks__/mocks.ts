import { Category, Product } from 'types';
export const mockCartItem: Product = {
	id: 1,
	title:
		'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
	price: 109.95,
	description:
		'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday',
	category: Category.MenSClothing,
	image:
		'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
	rating: {
		rate: 3.9,
		count: 120,
	},
};

export const mockSemiEmptyList: Product[] = [
	{
		id: 4,
		title: 'Mens Casual Slim Fit',
		price: 15.99,
		description:
			'The color could be slightly different between on the screen and in practice. / Please note that body builds vary by person, therefore, detailed size information should be reviewed below on the product description.',
		category: Category.MenSClothing,
		image:
			'https://fakestoreapi.com/img/71YXzeOuslL._AC_UY879_.jpg',
		rating: {
			rate: 2.1,
			count: 430,
		},
	},
];
export const mockList: Product[] = [
	{
		id: 4,
		title: 'Mens Casual Slim Fit',
		price: 15.99,
		description:
			'The color could be slightly different between on the screen and in practice. / Please note that body builds vary by person, therefore, detailed size information should be reviewed below on the product description.',
		category: Category.MenSClothing,
		image:
			'https://fakestoreapi.com/img/71YXzeOuslL._AC_UY879_.jpg',
		rating: {
			rate: 2.1,
			count: 430,
		},
	},
	{
		id: 5,
		title:
			"John Hardy Women's Legends Naga Gold & Silver Dragon Station Chain Bracelet",
		price: 695,
		description:
			"From our Legends Collection, the Naga was inspired by the mythical water dragon that protects the ocean's pearl. Wear facing inward to be bestowed with love and abundance, or outward for protection.",
		category: Category.Jewelery,
		image:
			'https://fakestoreapi.com/img/71pWzhdJNwL._AC_UL640_QL65_ML3_.jpg',
		rating: {
			rate: 4.6,
			count: 400,
		},
	},
	{
		id: 6,
		title: 'Solid Gold Petite Micropave ',
		price: 168,
		description:
			'Satisfaction Guaranteed. Return or exchange any order within 30 days.Designed and sold by Hafeez Center in the United States. Satisfaction Guaranteed. Return or exchange any order within 30 days.',
		category: Category.Jewelery,
		image:
			'https://fakestoreapi.com/img/61sbMiUnoGL._AC_UL640_QL65_ML3_.jpg',
		rating: {
			rate: 3.9,
			count: 70,
		},
	},
	{
		id: 7,
		title: 'White Gold Plated Princess',
		price: 9.99,
		description:
			"Classic Created Wedding Engagement Solitaire Diamond Promise Ring for Her. Gifts to spoil your love more for Engagement, Wedding, Anniversary, Valentine's Day...",
		category: Category.Jewelery,
		image:
			'https://fakestoreapi.com/img/71YAIFU48IL._AC_UL640_QL65_ML3_.jpg',
		rating: {
			rate: 3,
			count: 400,
		},
	},
	{
		id: 8,
		title:
			'Pierced Owl Rose Gold Plated Stainless Steel Double',
		price: 10.99,
		description:
			'Rose Gold Plated Double Flared Tunnel Plug Earrings. Made of 316L Stainless Steel',
		category: Category.Jewelery,
		image:
			'https://fakestoreapi.com/img/51UDEzMJVpL._AC_UL640_QL65_ML3_.jpg',
		rating: {
			rate: 1.9,
			count: 100,
		},
	},
];
