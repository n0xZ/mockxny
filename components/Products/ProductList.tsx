import { FC } from 'react';
import { Product } from 'types';
import ProductItem from './ProductItem';
interface ProductListProps {
	products: Product[];
}

const ProductList: FC<ProductListProps> = ({
	products,
}) => {
	return (
		<article className='grid xl:grid-cols-4 grid-cols-2 container mx-auto mt-6'>
			
			{products.map((product) => (
				<ProductItem product={product} key={product.id} />
			))}
		</article>
	);
};

export default ProductList;
