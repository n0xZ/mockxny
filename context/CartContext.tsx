import {
	createContext,
	FC,
	useContext,
	useState,
} from 'react';
import { useRouter } from 'next/router';
import { CartContextI, Product } from '../types';
import { toast } from 'react-hot-toast';
// Cart provider initialized, with its values based on the CartContextI interface
export const CartProvider = createContext<CartContextI>({
	clientCredit: 0,
	selectedProducts: [],
	addSelectedProduct: () => {},
	removeSelectedProduct: () => {},
	updatePaymentStates: () => {},
	getTotalPrice: () => 0,
	performPayment: () => {},
});

export const CartContext: FC = ({ children }) => {
	const router = useRouter();

	const [clientCredit, setClientCredit] = useState(1000);
	const [selectedProducts, setSelectedProducts] = useState<
		Product[]
	>([]);
	const addSelectedProduct = (product: Product) => {
		toast.success('Producto agregado al carrito');
		setSelectedProducts([...selectedProducts, product]);
	};
	const removeSelectedProduct = (index: number) => {
		setSelectedProducts(
			selectedProducts.filter((p, ind) => ind !== index)
		);
	};
	const getTotalPrice = () => {
		return Math.round(
			selectedProducts.reduce(
				(prevValue, currentValue) =>
					prevValue + currentValue.price,
				0
			)
		);
	};
	const updatePaymentStates = () => {
		setClientCredit(clientCredit - getTotalPrice());
		setSelectedProducts([]);
	};
	const performPayment = () => {
		if (getTotalPrice() < clientCredit) {
			router.push('/cart/payment/successful');
		} else {
			router.push('/cart/payment/error');
		}
	};
	return (
		<CartProvider.Provider
			value={{
				clientCredit,
				selectedProducts,
				addSelectedProduct,
				updatePaymentStates,
				removeSelectedProduct,
				getTotalPrice,
				performPayment,
			}}
		>
			{children}
		</CartProvider.Provider>
	);
};

export const useCard = () => useContext(CartProvider);
