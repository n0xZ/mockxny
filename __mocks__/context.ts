import { CartContextI } from 'types';
import { mockList } from './mocks';
export const mockDefaultValues: CartContextI = {
	clientCredit: 0,
	selectedProducts: [],
	addSelectedProduct: () => {},
	getTotalPrice: () => 0,
	updatePaymentStates: () => {},
	performPayment: () => {},
	removeSelectedProduct: () => {},
};

export const mockedSelectedProductValues: CartContextI = {
	clientCredit: 0,
	selectedProducts: mockList,
	addSelectedProduct: () => {},
	getTotalPrice: () => 0,
	updatePaymentStates: () => {},
	performPayment: () => {},
	removeSelectedProduct: () => {},
};
