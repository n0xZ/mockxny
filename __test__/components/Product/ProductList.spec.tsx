import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { mockList } from '@/__mocks__/mocks';
import ProductList from '@/components/Products/ProductList';
import { CartContext } from '@/context/CartContext';
describe('Product list component test case', () => {
	it('Renders the component without any trouble', () => {
		const { container } = render(
			<CartContext>
				<ProductList products={mockList} />
			</CartContext>
		);
		expect(container).toBeInTheDocument();
	});
	it('Displays the product items component based on the mock list', () => {
		const { container } = render(
			<ProductList products={mockList} />
		);
		expect(container).toMatchSnapshot();
	});
});
