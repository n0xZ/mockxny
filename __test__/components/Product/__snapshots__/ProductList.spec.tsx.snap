// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list component test case Displays the product items component based on the mock list 1`] = `
<div>
  <article
    class="grid xl:grid-cols-4 grid-cols-2 container mx-auto mt-6"
  >
    <aside
      class="flex flex-col justify-between p-3 hover:opacity-80 duration-100 ease-in-out"
    >
      <span
        class="flex flex-col"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Mens Casual Slim Fit"
            class="object-contain"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71YXzeOuslL._AC_UY879_.jpg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71YXzeOuslL._AC_UY879_.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71YXzeOuslL._AC_UY879_.jpg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
        <p
          class="font-bold mb-2"
        >
          Mens Casual Slim Fit
        </p>
      </span>
      <div
        class="w-full"
      >
        <p
          class="mb-3 text-gray-400 text-base relative"
        >
          $ 15.99
        </p>
        <button
          class="w-full border-2 font-lato text-base border-black rounded-md px-6 py-4 hover:border-gray-900 hover:bg-gray-200  duration-100 ease-in"
        >
          Agregar al carrito
        </button>
      </div>
    </aside>
    <aside
      class="flex flex-col justify-between p-3 hover:opacity-80 duration-100 ease-in-out"
    >
      <span
        class="flex flex-col"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="John Hardy Women's Legends Naga Gold & Silver Dragon Station Chain Bracelet"
            class="object-contain"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71pWzhdJNwL._AC_UL640_QL65_ML3_.jpg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71pWzhdJNwL._AC_UL640_QL65_ML3_.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71pWzhdJNwL._AC_UL640_QL65_ML3_.jpg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
        <p
          class="font-bold mb-2"
        >
          John Hardy Women's Legends Naga Gold & Silver Dragon Station Chain Bracelet
        </p>
      </span>
      <div
        class="w-full"
      >
        <p
          class="mb-3 text-gray-400 text-base relative"
        >
          $ 695
        </p>
        <button
          class="w-full border-2 font-lato text-base border-black rounded-md px-6 py-4 hover:border-gray-900 hover:bg-gray-200  duration-100 ease-in"
        >
          Agregar al carrito
        </button>
      </div>
    </aside>
    <aside
      class="flex flex-col justify-between p-3 hover:opacity-80 duration-100 ease-in-out"
    >
      <span
        class="flex flex-col"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Solid Gold Petite Micropave "
            class="object-contain"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F61sbMiUnoGL._AC_UL640_QL65_ML3_.jpg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F61sbMiUnoGL._AC_UL640_QL65_ML3_.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F61sbMiUnoGL._AC_UL640_QL65_ML3_.jpg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
        <p
          class="font-bold mb-2"
        >
          Solid Gold Petite Micropave 
        </p>
      </span>
      <div
        class="w-full"
      >
        <p
          class="mb-3 text-gray-400 text-base relative"
        >
          $ 168
        </p>
        <button
          class="w-full border-2 font-lato text-base border-black rounded-md px-6 py-4 hover:border-gray-900 hover:bg-gray-200  duration-100 ease-in"
        >
          Agregar al carrito
        </button>
      </div>
    </aside>
    <aside
      class="flex flex-col justify-between p-3 hover:opacity-80 duration-100 ease-in-out"
    >
      <span
        class="flex flex-col"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="White Gold Plated Princess"
            class="object-contain"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71YAIFU48IL._AC_UL640_QL65_ML3_.jpg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71YAIFU48IL._AC_UL640_QL65_ML3_.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F71YAIFU48IL._AC_UL640_QL65_ML3_.jpg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
        <p
          class="font-bold mb-2"
        >
          White Gold Plated Princess
        </p>
      </span>
      <div
        class="w-full"
      >
        <p
          class="mb-3 text-gray-400 text-base relative"
        >
          $ 9.99
        </p>
        <button
          class="w-full border-2 font-lato text-base border-black rounded-md px-6 py-4 hover:border-gray-900 hover:bg-gray-200  duration-100 ease-in"
        >
          Agregar al carrito
        </button>
      </div>
    </aside>
    <aside
      class="flex flex-col justify-between p-3 hover:opacity-80 duration-100 ease-in-out"
    >
      <span
        class="flex flex-col"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Pierced Owl Rose Gold Plated Stainless Steel Double"
            class="object-contain"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F51UDEzMJVpL._AC_UL640_QL65_ML3_.jpg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F51UDEzMJVpL._AC_UL640_QL65_ML3_.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Ffakestoreapi.com%2Fimg%2F51UDEzMJVpL._AC_UL640_QL65_ML3_.jpg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
        <p
          class="font-bold mb-2"
        >
          Pierced Owl Rose Gold Plated Stainless Steel Double
        </p>
      </span>
      <div
        class="w-full"
      >
        <p
          class="mb-3 text-gray-400 text-base relative"
        >
          $ 10.99
        </p>
        <button
          class="w-full border-2 font-lato text-base border-black rounded-md px-6 py-4 hover:border-gray-900 hover:bg-gray-200  duration-100 ease-in"
        >
          Agregar al carrito
        </button>
      </div>
    </aside>
  </article>
</div>
`;
